packages/evm/contracts/payments/PaymentsBook.sol

```diff
@@ -351,8 +306,8 @@ contract PaymentsBook is ControlledRecoverable, TimeHelpers, IPaymentsBook {
     )
         internal
     {
+        _period.claimedGuardians[_guardian][_token] = true;
         if (_amount > 0) {
```

Comment:
After a protracted discussion on the pros and cons of this (unfortunately adding this check above was not so easy because we may need to obtain the randomness for fetching a guardian's balance in any given period), we ultimately decided to re-flip the getter so that it reports whether or not a guardian has already claimed a token.

UIs and contracts will be able to use this getter to know whether a claim request has already been processed, and not request it again. It should be safe, as claim requests resulting in 0 tokens being transferred are still allowed.